trace-utils/src/tracer_payload.rs

```diff
@@ -181,4 +304,95 @@ mod tests {
             panic!("Unexpected type");
         }
     }
+
+    #[test]
+    fn test_try_into_success() {
+        let span_data1 = json!([{
+            "service": "test-service",
+            "name": "test-service-name",
+            "resource": "test-service-resource",
+            "trace_id": 111,
+            "span_id": 222,
+            "parent_id": 100,
+            "start": 1,
+            "duration": 5,
+            "error": 0,
+            "meta": {},
+            "metrics": {},
+        }]);
+
+        let expected_serialized_span_data1 = vec![Span {
+            service: "test-service".to_string(),
+            name: "test-service-name".to_string(),
+            resource: "test-service-resource".to_string(),
+            trace_id: 111,
+            span_id: 222,
+            parent_id: 100,
+            start: 1,
+            duration: 5,
+            error: 0,
+            meta: HashMap::new(),
```

Comment:
Not related to this PR. But we whould generally avoid using Hashmaps for data which doesn't require it.

e.g. in many tracers we replaced HashMap for meta and metrics with a list of key,values - for internal representation - because it avoid a lot of unnecessary memory operations. And is generally alot faster since the map is very rarely looked up.

Reply:
I don't know about that, but at least PHP does quite a bunch of meta/metrics lookups at serialization time.
Also, e.g. sampling can be tag based.
And there are also quite a bunch of tags which are tied to state, like sampling, decision makers etc. which end up requiring lookups.
I doubt that a key-value list would bring tangible benefits over a solid HashTable impl.